client/src/app/routes/stardust/Foundry.tsx

```diff
@@ -0,0 +1,172 @@
+import { IAliasAddress, IFoundryOutput, IImmutableAliasUnlockCondition } from "@iota/iota.js-stardust";
+import React, { useContext, useEffect, useState } from "react";
+import { RouteComponentProps } from "react-router";
+import { ServiceFactory } from "../../../factories/serviceFactory";
+import { ClipboardHelper } from "../../../helpers/clipboardHelper";
+import { formatAmount } from "../../../helpers/stardust/valueFormatHelper";
+import { STARDUST } from "../../../models/db/protocolVersion";
+import { StardustTangleCacheService } from "../../../services/stardust/stardustTangleCacheService";
+import CopyButton from "../../components/CopyButton";
+import FiatValue from "../../components/FiatValue";
+import Icon from "../../components/Icon";
+import NetworkContext from "../../context/NetworkContext";
+import { FoundryProps } from "./FoundryProps";
+import "./Foundry.scss";
+
+const Foundry: React.FC<RouteComponentProps<FoundryProps>> = ({ match: { params: { network, outputId } } }) => {
+    const { tokenInfo } = useContext(NetworkContext);
+    const [serialNumber, setSerialNumber] = useState<number>();
+    const [controllerAlias, setControllerAlias] = useState<string>();
+    const [balance, setBalance] = useState<number>();
+    const [tokenScheme, setTokenScheme] = useState<number>();
+    const [maximumSupply, setMaximumSupply] = useState<number>();
+    const [mintedTokens, setMintedTokens] = useState<number>();
+    const [meltedTokens, setMeltedTokens] = useState<number>();
+
+    useEffect(() => {
+        const tangleCacheService = ServiceFactory.get<StardustTangleCacheService>(`tangle-cache-${STARDUST}`);
+        tangleCacheService.outputDetails(network, outputId).then(outputDetails => {
+            const foundryOutput = outputDetails?.output as IFoundryOutput;
+            const immutableAliasUnlockCondition = (foundryOutput.unlockConditions[0] as IImmutableAliasUnlockCondition);
+            const aliasId = (immutableAliasUnlockCondition.address as IAliasAddress).aliasId;
+
+            setSerialNumber(foundryOutput.serialNumber);
+            setControllerAlias(aliasId);
+            setBalance(Number(foundryOutput.amount));
+            setTokenScheme(foundryOutput.tokenScheme.type);
+            setMaximumSupply(Number(foundryOutput.tokenScheme.maximumSupply));
+            setMintedTokens(Number(foundryOutput.tokenScheme.mintedTokens));
+            setMeltedTokens(Number(foundryOutput.tokenScheme.meltedTokens));
+        })
+        .catch(_ => {});
```

Comment:
Should this be default to something in case the API fails?

Reply:
This is in `then` of the async request, so the request should be successful at this point.